xclim/sdba/adjustment.py

```diff
@@ -215,6 +217,29 @@ class DetrendedQuantileMapping(EmpiricalQuantileMapping):
     .. [Cannon2015] Cannon, A. J., Sobie, S. R., & Murdock, T. Q. (2015). Bias correction of GCM precipitation by quantile mapping: How well do methods preserve changes in quantiles and extremes? Journal of Climate, 28(17), 6938–6959. https://doi.org/10.1175/JCLI-D-14-00754.1
     """
 
+    @parse_group
+    def __init__(
+        self,
+        *,
+        nquantiles: int = 20,
+        kind: str = ADDITIVE,
+        interp: str = "nearest",
+        detrending: Union[int, BaseDetrend] = 1,
+        extrapolation: str = "constant",
+        group: Union[str, Grouper] = "time",
+    ):
+
+        super().__init__(
+            nquantiles=nquantiles,
+            kind=kind,
+            interp=interp,
+            extrapolation=extrapolation,
+            group=group,
+        )
+        if isinstance(detrending, int):
+            detrending = PolyDetrend(degree=detrending, kind=self.kind)
+        self["detrending"] = detrending
```

Comment:
A bit weird. Would use `self.detrending` instead.

Reply:
And why do we need this in the init ?

Reply:
If I use `self.detrending` it won't be "registered" as a parameter, that was my point with flushing the  `__setattr__ = dict.__setitem__` part.
It is in the init so we can use  `super().__init__` without having to define a meaningless `**kwargs` in `EmpiricalQuantileMapping`.

Reply:
But why does it need to be initialized at all if it is an argument of adjust ?